numba/tests/test_ufuncs.py

```diff
@@ -1178,12 +1191,11 @@ class TestScalarUFuncs(TestCase):
     This class tests regular uses. A subclass tests the no python backend.
     """
 
-    _compile_flags = enable_pyobj_flags
+    _nopython = False
```

Comment:
Whilst this is just a refactor, I'm puzzled by the use of `enable_pyobj_flags`/`nopython=False`. IIRC these flags essentially permit "fallback" to `object mode` should the `nopython mode` pipeline fail in compilation, but everything being tested looks like it would succeed under `nopython mode`, this is tested again in `TestScalarUFuncsNoPython` where `nopython mode` is enforced? Is there any need for this?

Reply:
It's been this way since the tests were added in 2014, in 7c20a342cde38793a735382e033d17f190a49291. There's probably no need for it now, but I can see that back in 2014 it might have been handy to be able to tell if nopython mode stopped working whilst object mode succeeded. Do you want this removing and all tests to use just nopython mode always? Or shall we leave test semantics as they are, as this is just a refactor to get towards being able to remove `compiled_isolated` / `NestedContext`, and use the ufunc tests for CUDA ufunc testing?